lib/tpm2_policy.c

```diff
@@ -200,6 +200,92 @@ bool tpm2_policy_build_pcr(TSS2_SYS_CONTEXT *sapi_context,
     return result;
 }
 
+bool tpm2_policy_build_policyauthorize(TSS2_SYS_CONTEXT *sapi_context,
+    tpm2_session *policy_session, const char *policy_input,
+    const char *policy_qualifier, const char *public_key_name, const char *ticket) {
+
+    unsigned long file_size = 0;
+
+    bool result = files_get_file_size_path(policy_input, &file_size);
+    if (!result) {
+        LOG_ERR("Could not retrieve file size of the policy to authorize");
+        return false;
+    }
+
+    TPM2B_DIGEST approvedPolicy = {
+        .size = (uint16_t)file_size
+    };
+    result = files_load_bytes_from_path(policy_input,
+        approvedPolicy.buffer, &approvedPolicy.size);
+    if (!result) {
+        LOG_ERR("Could not retrieve policy to authorize from file");
+        return false;
+    }
+
+    result = files_get_file_size_path(policy_qualifier,
+        &file_size);
+    if (!result) {
+        LOG_ERR("No policy qualifier data given, choosing blank");
+        file_size = 0;
+    }
+    TPM2B_NONCE policy_ref = {
+        .size = (uint16_t) file_size
+    };
+
+    switch(file_size) {
+        case 0:
+            break;
+        default:
+            result = files_load_bytes_from_path(policy_qualifier,
+                policy_ref.buffer, &policy_ref.size);
+            if (!result) {
+                LOG_ERR("Could not retrieve policy reference to authorize from file");
+                return false;
+            }
+    }
+
+    result = files_get_file_size_path(public_key_name, &file_size);
+    if (!result) {
+        LOG_ERR("Could not retrieve file size of the verifying public key name");
+        return false;
+    }
+    TPM2B_NAME keySign = {
+        .size = (uint16_t)file_size
+    };
+
+    result = files_load_bytes_from_path(public_key_name, keySign.name,
```

Comment:
Is size 0 an error here?

Reply:
Im not talking about result. On line 247 you get the file size, is it ok for it to be 0?

files_load_bytes_from_path() will gladly load 0 bytes from a file AFAICT.